docs/devel/export-map.rst

```diff
@@ -0,0 +1,158 @@
+
+``-v``, ``--variant``
+    The specific variant of a tile that have visual variations. [number] (like
+    grass tiles in ASCII mode)
+
+``-h``, ``--hidden``
+    Whether tile is revealed or unrevealed. [boolean]
+
+``-l``, ``--light``
+    Whether tile is exposed to light. [boolean]
+
+``-b``, ``--subterranean``
+    Whether the tile is considered underground. [boolean] (used to determine
+    crops that can be planted underground)
+
+``-o``, ``--outside``
+    Whether the tile is considered “outside”. [boolean] (used by weather effects
+    to trigger on outside tiles)
+
+``-a``, ``--aquifer``
+    Whether the tile is considered an aquifer. [number ID] (NONE/LIGHT/HEAVY)
+
+``-m``, ``--material``
+    The material inside the tile. [number ID] (IRON/GRANITE/CLAY/
+    TOPAZOLITE/BLACK_OPAL/etc.) (will return nil if the tile is empty)
+
+``-q``, ``--liquid``
+    The type of liquid inside the tile. [number ID] (WATER/MAGMA) (will return
+    nil if the tile flow level is zero)
+
+``-f``, ``--flow``
+    The level of liquids inside the tile. [number] (0-7)
+
+``-u``, ``--underworld``
+    Whether the underworld z-levels will be included.
+
+``-e``, ``--evilness``
+    Whether the evilness value will be included in MAP_SIZE table. This only
+    checks the value of the center map tile at ground level and will ignore
+    biomes at the edges of the map.
+
+JSON DATA
+---------
+
+``ARGUMENT_OPTION_ORDER``
+    The order of the selected options for how data is arranged at a map
+    position.
+
+    Example 1:
+        ``{"material": 1, "shape": 2, "hidden": 3}``
+
+        ``map[z][y][x] = {material_data, shape_data, hidden_data}``
+
+    Example 2:
+        ``{"variant": 3, "light": 1, "outside": 2, "aquifer": 4}``
+
+        ``map[z][y][x] = {light_data, outside_data, variant_data, aquifer_data}``
+
+``MAP_SIZE``
+    A table containing basic information about the map size for width, height,
+    depth. (x, y, z) The underworld_z_level is included if the underworld option
+    is enabled and the map depth (z) will be automatically adjusted.
+
+``KEYS``
+    The tables containing the [number ID] values for different options.
+
+    ``"SHAPE": {"-1": "NONE", "0": "EMPTY", "1": "FLOOR", "2": "BOULDERS",
+    "3": "PEBBLES", "4": "WALL", ..., "18": "ENDLESS_PIT"}``
+
+    ``"PLANT": {"0": "SINGLE-GRAIN_WHEAT", "1": "TWO-GRAIN_WHEAT",
+    "2": "SOFT_WHEAT", "3": "HARD_WHEAT", "4": "SPELT", "5": "BARLEY", ...,
+    "224": "PALM"}``
+
+    ``"AQUIFER": {"0": "NONE", "1": "LIGHT", "2": "HEAVY"}``
+
+    Note - when using the ``materials`` option, you need to pair the [number ID]
+    with the correct ``KEYS`` material table. Generally you use ``tiletype``
+    option as a helper to sort tiles into different material types. I would
+    recommend consulting ``tile-material.lua`` to see how materials are sorted.
+
+``map``
+    JSON map data is arranged as: ``map[z][y][x] = {tile_data}``
+
+    JSON maps start at index [1]. (starts at map[1][1][1])
```

Comment:
I don't think this is true. It's **Lua** lists that start at [1]. JSON arrays are zero-indexed.

Reply:
Interesting. 

The JSON library I was using in DM to import the data counts the arrays starting at 1. They probably did that for compatibility reasons since DM arrays start at [1]. Thought it was universal.

Let me tweak the wording to reference that.

Reply:
Applied via https://github.com/DFHack/scripts/pull/1329/commits/9338977b7965c5a5d13020ddc1d4bd555affce95